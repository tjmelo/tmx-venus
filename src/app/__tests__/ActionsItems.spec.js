import toActions from '../modules/ActionsItems';

const mockElementID = {
  id: 10,
  element: 'some',
};

describe('Should verify if the toActions function receives a determinated params', () => {
  it('Should mount structure action for item', () => {
    const testActions = jest.fn();

    testActions.mockImplementationOnce((argTest1) => toActions(argTest1));

    // do
    testActions(mockElementID.id);

    // then
    expect(testActions).toHaveReturned();
    expect(testActions).toHaveBeenCalledTimes(1);
    expect(testActions).toHaveBeenCalledWith(mockElementID.id);

    // Expects fidelity with scope of function
    expect(testActions).toMatchSnapshot();
  });
});
