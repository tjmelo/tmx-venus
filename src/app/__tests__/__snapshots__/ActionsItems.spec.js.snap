// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should verify if the toActions function receives a determinated params Should mount structure action for item 1`] = `
[MockFunction] {
  "calls": [
    [
      10,
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": <span
        class="box-actions"
      >
        <button>
          <i
            class="d-none fas fa-cart-arrow-down"
            id="10"
            title="Save"
          />
        </button>
        <button>
          <i
            class="fas fa-pen-square"
            id="10"
            title="Edit"
          />
        </button>
        <button>
          <i
            class="fas fa-trash"
            id="10"
            title="Delete"
          />
        </button>
      </span>,
    },
  ],
}
`;
